### PR TITLE
LPS-85850 New Indexers: BookmarksEntry & BookmarksFolder

### PR DESCRIPTION
**Pull request by ifnazar**

### CODE REVIEW

**Comment by ifnazar** [comment]
ci:test:sf

**Comment by ifnazar** [comment]
ci:test:search

File: modules/apps/bookmarks/bookmarks-service/src/main/java/com/liferay/bookmarks/internal/search/BookmarksEntryModelIndexerWriterContributor.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+				dynamicQuery.add(statusProperty.in(statuses));
+			});
+
+		batchIndexingActionable.setPerformActionMethod(
+			(BookmarksEntry bookmarksEntry) -> {
+				batchIndexingActionable.addDocuments(
+					modelIndexerWriterDocumentHelper.getDocument(
+						bookmarksEntry));
+
+				bookmarksFolderBatchReindexer.reindex(
```
Comment (luanmaoski) [inline modules/apps/bookmarks/bookmarks-service/src/main/java/com/liferay/bookmarks/internal/search/BookmarksEntryModelIndexerWriterContributor.java]: Hi @brandizzi.

On `BookmarksEntryIndexer` we have this code:

` Property folderIdProperty = PropertyFactoryUtil.forName(
	"folderId");
     dynamicQuery.add(folderIdProperty.eq(folderId)); `

We created the `BookmarksFolderBatchReindexer` to call a reindex passing the `folderId` as a parameter, but we are not sure if this is the correct behavior.

Is there a away to get the `folderId` inside the `setAddCriteriaMethod` of this class?

**Comment by ifnazar** [comment]
ci:test:relevant

File: modules/apps/bookmarks/bookmarks-service/src/main/java/com/liferay/bookmarks/internal/search/BookmarksEntryModelIndexerWriterContributor.java (hunk 1 above)
Comment (brandizzi) [reply modules/apps/bookmarks/bookmarks-service/src/main/java/com/liferay/bookmarks/internal/search/BookmarksEntryModelIndexerWriterContributor.java]: @luanmaoski I read the code of `DefaultIndexer` and `IndexerWriterImpl` here and apparently you're right. I guess you'll have to implement a `ModelIndexerWriterContributor` for bookmarks with a `getBatchIndexingActionable()` method. Do you foresee any issues? If so, let me know!

Comment (luanmaoski) [reply modules/apps/bookmarks/bookmarks-service/src/main/java/com/liferay/bookmarks/internal/search/BookmarksEntryModelIndexerWriterContributor.java]: Alright. Thanks.

**Comment by ifnazar** [comment]
ci:test:sf

**Comment by ifnazar** [comment]
ci:test:search

**Comment by ifnazar** [comment]
ci:test:search

**Comment by ifnazar** [comment]
ci:test:sf

**Comment by ifnazar** [comment]
ci:test:sf

**Comment by ifnazar** [comment]
ci:test:search

**Comment by brandizzi** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by brandizzi** [comment]
Pull request submitted to https://github.com/brianchandotcom/liferay-portal/pull/66807. <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Pull request closed without merging**